### ISSUE TITLE
mark: Resets the filter upon execution

### ISSUE THREAD

**Issue by nus-se-script** [nus-cs2103-AY2425S1/pe-dev-response#207]
After marking an assignment, the `find` filter is reset.

Steps to reproduce (starting from a fresh install):
1. Create an assignment: `add_assignment si/1 an/Test Assignment ms/100`
2. Filter the first person: `find Alex`
3. Delete the assignment: `mark si/1 ai/1`
4. Note that the `find` filter is reset

After step 2:
![Screenshot 2024-11-15 at 5.08.51 PM.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/05a4507b-350d-46a3-9988-a7c99064c9cf.png)

After step 3 (Actual):
![Screenshot 2024-11-15 at 5.19.48 PM.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/a6078f94-2bcc-4677-a622-717345c5439b.png)

After step 3 (Expected):
Note that the command output should be the same as the screenshot above. Other than that, the screenshot below is accurate.
![Screenshot 2024-11-15 at 5.08.51 PM.png](https://raw.githubusercontent.com/nus-cs2103-AY2425S1/pe/master/files/05a4507b-350d-46a3-9988-a7c99064c9cf.png)

<hr>
<sub>[original: nus-cs2103-AY2425S1/pe-interim#292] [original labels: severity.Medium type.FunctionalityBug]</sub>

**Comment by emmannyyy** [comment]
# Team's Response
This is essentially a duplicate, where the find filter is intentionally reset after each command.

It was intentional behaviour as we believe that users would want to see the main student list after executing each command.


## Duplicate status (if any):
Duplicate of #1656
